python/pyspark/sql/dataframe.py

```diff
@@ -2536,6 +2536,28 @@ def withColumnRenamed(self, existing, new):
         """
         return DataFrame(self._jdf.withColumnRenamed(existing, new), self.sql_ctx)
 
+    def withMetadata(self, columnName, metadata):
+        """Returns a new :class:`DataFrame` by updating an existing column with metadata.
+
+        .. versionadded:: 3.3.0
+
+        Parameters
+        ----------
+        columnName : str
+            string, name of the existing column to update the metadata.
+        metadata : dict
+            dict, new metadata to be assigned to df.schema[columnName].metadata
+
+        Examples
+        --------
+        >>> df_meta = df.withMetadata('age', {'foo': 'bar'})
+        >>> df_meta.schema['age'].metadata
+        {'foo': 'bar'}
+        """
+        if not isinstance(metadata, dict):
+            raise TypeError("metadata should be a dict")
+        return DataFrame(self._jdf.withMetadata(columnName, metadata), self.sql_ctx)
```

Comment:
The metadata argument type is a dict , how does it pass to the java method `def withMetadata(columnName: String, metadata: Metadata)` and the metadata become a `Metadata` class type ?

Reply:
Nice catch! I updated the conversion.